metacluster/include/metacluster/ListTenants.actor.h

```diff
@@ -130,14 +162,14 @@ Future<std::vector<std::pair<TenantName, MetaclusterTenantMapEntry>>> listTenant
 			state int count = 0;
 			loop {
 				std::vector<std::pair<TenantName, MetaclusterTenantMapEntry>> tenantBatch =
-				    wait(listTenantMetadataTransaction(tr, begin, end, std::max(limit + offset, 1000)));
+				    wait(listTenantMetadataTransaction(tr, begin, end, std::max(limit + offset, 1000), tenantGroup));
 
 				if (tenantBatch.empty()) {
 					return results;
 				}
 
 				for (auto const& [name, entry] : tenantBatch) {
-					if (std::count(filters.begin(), filters.end(), entry.tenantState)) {
+					if (filters.empty() || std::count(filters.begin(), filters.end(), entry.tenantState)) {
```

Comment:
We shouldn't get here if `filters.empty` is true because of the block at line 150